lib/order.rb

```diff
@@ -0,0 +1,68 @@
+require "csv"
+require_relative "customer"
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    if (@fulfillment_status != :pending && @fulfillment_status != :shipped && @fulfillment_status != :paid && @fulfillment_status != :processing && @fulfillment_status != :complete)
+      raise ArgumentError, "That is not a valid fulfillment status."
+    end
+  end
+
+  def total
+    sum = 0
+    @products.each_value do |price|
+      sum += price
+    end
+    sum_with_tax = sum + (sum * 0.075)
+    return sum_with_tax.round(2)
+  end
+
+  def add_product(name, price)
+    if @products.include?(name)
+      raise ArgumentError, "This product has already been added to the order."
+    end
+    @products[name] = [price]
+  end
+
+  # optional method to remove product by name
+  def remove_product(product_name)
+    if @products.include?(product_name)
+      @products.delete(product_name)
+    else
+      raise ArgumentError, "There is no product with that name."
+    end
+  end
+
+  def self.parse_helper(product_string)
+    array_strings = product_string.split(";")
+    item_hash = Hash[array_strings.map { |item| item.split ":" }]
+    item_hash.each do |key, value|
+      item_hash[key] = value.to_f
+    end
+  end
```

Comment:
Personally, I'd probably prefer an explicit return statement for this method to make it clear what you're giving back